libs/block-mesh-manager/src/database/ip_address/get_or_create_ip_address.rs

```diff
@@ -0,0 +1,43 @@
+use crate::domain::ip_address::IpAddress;
+use chrono::Utc;
+use sqlx::{Postgres, Transaction};
+use uuid::Uuid;
+
+pub async fn get_or_create_ip_address(
+    transaction: &mut Transaction<'_, Postgres>,
```

Comment:
I'd suggest using something more generic than Transaction for a DB connection type. Including every query in transaction is not always necessary.